### PR TITLE
Deep conus evidential notebook and figures

### PR DESCRIPTION
**Pull request by djgagne**
Here is the code used for the ECMWF evidential XAI presentation. Most of my key code is in the notebooks/deepconus_ev.ipynb.

### CODE REVIEW

**Comment by jsschreck** [comment]
Do you want to commit the gifs and what not?

**Comment by charlie-becker** [comment]
> Do you want to commit the gifs and what not?

Sure. Do you think they are better suited here or in P-Type?

**Review by charlie-becker — APPROVED** [review]
It appears you return `belief` and the code I have from John returns `prob` as the first axis from the evidential output. I understand they are related, but which is appropriate to return? What is the relationship between them? Is `belief` == `prob - u`? Will the argmax of belief always return the correct class? Also, why is is `1` returned in addition to the list of parameters?

```
class Evidential(tf.keras.layers.Layer):
    def __init__(self, classes=2):
        super(Evidential, self).__init__()
        self.classes = classes
        
    def call(self, inputs):
        evidence = K.softplus(inputs)
        alphas = evidence + 1
        S = tf.reduce_sum(alphas, axis=1, keepdims=True)
        u = self.classes / S
        prob = alphas / S
        belief = evidence / S
        epistemic = prob * (1 - prob) / (S + 1)
        aleatoric = prob - prob ** 2 - epistemic
        merged = tf.concat([belief, u, epistemic, aleatoric], 1)
        return merged
```

Looks like the CI failed, but is not a result of this PR. Overall, looks good. The analysis is helpful.

**Comment by djgagne** [comment]
I would like to include the images as examples but will try not to add more and drag the weight of the repo down too much. I fixed the flake8 errors as well.

**Comment by djgagne** [comment]
> It appears you return `belief` and the code I have from John returns `prob` as the first axis from the evidential output. I understand they are related, but which is appropriate to return? What is the relationship between them? Is `belief` == `prob - u`? Will the argmax of belief always return the correct class? Also, why is is `1` returned in addition to the list of parameters?
> 
> ```
> class Evidential(tf.keras.layers.Layer):
>     def __init__(self, classes=2):
>         super(Evidential, self).__init__()
>         self.classes = classes
>         
>     def call(self, inputs):
>         evidence = K.softplus(inputs)
>         alphas = evidence + 1
>         S = tf.reduce_sum(alphas, axis=1, keepdims=True)
>         u = self.classes / S
>         prob = alphas / S
>         belief = evidence / S
>         epistemic = prob * (1 - prob) / (S + 1)
>         aleatoric = prob - prob ** 2 - epistemic
>         merged = tf.concat([belief, u, epistemic, aleatoric], 1)
>         return merged
> ```
> 
> Looks like the CI failed, but is not a result of this PR. Overall, looks good. The analysis is helpful.

The 1 in tf.concat is the axis of concatenation. The argmax of belief should return the correct class but may also return the uncertain class if it is the highest one, which is part of the reason to use belief instead of the raw probabilities.

**Comment by djgagne** [comment]
Merging now.
